Feature/co 665 advisor api

floppyears opened the pull request.

SuzyWu2014 committed:
update swagger

SuzyWu2014 committed:
update based on comments

SuzyWu2014 committed:
Merge pull request #1 from osu-mist/feature/co-664-spec-design

update swagger.yaml

floppyears committed:
Add basic healthcheck, resource and dao

CO-665

floppyears committed:
Improve safety of clone contrib

CO-665

floppyears committed:
Update .gitignore

CO-665

floppyears committed:
Add placeholder DAO, Mapper and wiring code

CO-665

Healthcheck uses DAO + mapper and application registers new components.
The next step is put the actual query in the contrib files and update
the mapper.

floppyears committed:
First stab at the advisors api

CO-665

Adding query to DAO
Adding DAO
The resource class returns a jsonapi result object

floppyears committed:
Update contrib repo sha1

CO-665

jaredkosanovic commented on src/main/groovy/edu/oregonstate/mist/advisors/resources/AdvisorsResource.groovy:
```diff
@@ -0,0 +1,51 @@
+        def resultObject = new ResultObject()
+        attributes.each {
+            resultObject.data += new ResourceObject(
+                    id:         1, //@todo: what do we use for an id?
```
I don't think this should be hard coded. JSONapi states that a id and type pair must be unique for each resource object. http://jsonapi.org/format/#document-resource-object-identification
Maybe we can use the advisor's OSU ID?

floppyears replied:
I agree. I just had it hard coded since I didn't have a good solution for it. We don't want to expose the osu id of the supervisor since that would require extra permissions / MOU. Also the advisor id is not unique since multiple students may have the same advisor. I'm leaning towards using a hashCode of the object as the id.

floppyears committed:
Refactor the api to simplify query

CO-665

Add hashCode to attributes so that we can have an id

floppyears committed:
Minor changes / clean up

CO-665

Update swagger to reflect a single id parameter.
Update contrib repo reference.
